doc/ADR/cpp_api_modularisation.md

```diff
@@ -0,0 +1,103 @@
+\page cpp-api-and-modularisation Libcprover-cpp and Modularisation
+
+**Date**: 22 Jun 2023
+**Updated**: 22 Jun 2023
+**Author**: Fotis Koutoulakis, fotis.koutoulakis@diffblue.com
+**Domain**: Architecture, API
+**Description**: This document outlines our thinking about the rearchitecting of
+    CBMC using the C++ API (`Libcprover-cpp`) as the central module and the
+    transitioning of other tools to use that as a basis.
+
+## Motivation && Current State of Affairs
+
+CProver is a collection of tools fitting various program analysis needs. It
+has been the product of the evolution of the codebase of the model-checking
+tool for C (`CBMC`), after it has been adopted with various front-ends/back-ends
+and various auxilliary tools.
+
+During this time, the repository has grown organically, using some guidelines
+for development that were based on tradition and intuition rather than some
+tight development guide. This development model has been successful for most
```

Comment:
"tight development guide" -> "agreed architectural approach"
This may not be the best choice either, but we want to be clear that there has not been a formal/agreed way to developer CProver and how to write code and use structures/modules within the codebase.